File: FileLibrary/src/main/java/com/xx/Filelibrary/ui/adapter/PicAdapter.java
package com.xx.Filelibrary.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.Transformation;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.request.BaseRequestOptions;
import com.bumptech.glide.request.RequestOptions;
import com.xx.Filelibrary.R;
import com.xx.Filelibrary.ui.adapter.viewHolder.OfficeViewHolder;
import com.xx.Filelibrary.ui.adapter.viewHolder.PicViewHolder;

import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.util.List;

import kotlin.jvm.internal.Intrinsics;

public class PicAdapter extends xxBaseRecycleViewAdapter<File, PicViewHolder> {


    public PicAdapter(List<File> list, Context context) {
        super(list, context);
    }

    @Override
    protected int getContentView(int viewType) {
        return R.layout.item_image;
    }

    @Override
    protected void covert(PicViewHolder holder, File data, int position) {
        RequestOptions options = RequestOptions.bitmapTransform(new RoundedCorners(40));
        Glide.with(mContext).load(data.getPath()).apply(options).into(holder.iv);
    }


}

File: FileLibrary/src/main/java/com/xx/Filelibrary/PicAdapter.kt
package com.xx.Filelibrary

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import java.io.File

class PicAdapter(c: Context, l: List<File>):RecyclerView.Adapter<PicViewHolder>() {
    val context:Context=c;
    val list:List<File> = l

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PicViewHolder {

        val view: View =LayoutInflater.from(context).inflate(R.layout.item_image, parent, false);

        return PicViewHolder(view);
    }

    override fun onBindViewHolder(holder: PicViewHolder, position: Int) {

        //Glide设置图片圆角角度
        val roundedCorners = RoundedCorners(40)
        //通过RequestOptions扩展功能,override:采样率,因为ImageView就这么大,可以压缩图片,降低内存消耗
        // RequestOptions options = RequestOptions.bitmapTransform(roundedCorners).override(20, 20);
        //通过RequestOptions扩展功能,override:采样率,因为ImageView就这么大,可以压缩图片,降低内存消耗
        // RequestOptions options = RequestOptions.bitmapTransform(roundedCorners).override(20, 20);
        val options: RequestOptions = RequestOptions.bitmapTransform(roundedCorners)
        Glide.with(context).load(list.get(position).getPath()).apply(options).into(holder.iv)
    }

    override fun getItemCount(): Int {
        return list.size
    }
}